store/sqlstore/team_store.go

```diff
@@ -282,6 +283,33 @@ func (s SqlTeamStore) GetByName(name string) (*model.Team, *model.AppError) {
 	return &team, nil
 }
 
+func (s SqlTeamStore) GetByNames(names []string) ([]*model.Team, *model.AppError) {
+	uniqueNames := utils.RemoveDuplicatesFromStringArray(names)
+	teams := []*model.Team{}
+
+	query := s.getQueryBuilder().
+		Select("*").
+		From("Teams").
+		Where(sq.Eq{"Name": uniqueNames})
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return nil, model.NewAppError("SqlTeamStore.GetByNames", "store.sql_team.get_by_names.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	_, err = s.GetReplica().Select(&teams, queryString, args...)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, model.NewAppError("SqlTeamStore.GetByNames", "store.sql_team.get_by_names.missing.app_error", nil, err.Error(), http.StatusNotFound)
+		}
+		return nil, model.NewAppError("SqlTeamStore.GetByNames", "store.sql_team.get_by_names.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	if len(teams) != len(uniqueNames) {
+		return nil, model.NewAppError("SqlTeamStore.GetByNames", "store.sql_team.get_by_names.missing.app_error", nil, "", http.StatusNotFound)
```

Comment:
Nit: maybe the error could be replaced with `409 - Conflict`

Reply:
I don't think is a 409 error here, I'm following the "all or nothing" approach here, if you try to get multiple teams by name and there is one that doesn't exists, it gets a 404, because you are looking for something that is not found.